### PR TITLE
feat[snippets]: gen1 React SDK snippets (Next.js, Remix, React)

### PR DESCRIPTION
**Pull request by sidmohanty11**
## Description

This PR adds gen1 React SDK snippets (Next.js, Remix, React)

_Screenshot_
If relevant, add a screenshot or two of the changes you made.

### CODE REVIEW

File: packages/sdks/snippets/gen1-remix/app/routes/_index.tsx [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export * from '~/routes/$';
```
Comment (sidmohanty11) [inline packages/sdks/snippets/gen1-remix/app/routes/_index.tsx:1]: https://github.com/remix-run/remix/discussions/5362

Comment (sidmohanty11) [reply packages/sdks/snippets/gen1-remix/app/routes/_index.tsx]: faced build errors with ^ so used: https://github.com/remix-run/remix/issues/5841#issuecomment-1526376558

File: packages/sdks/snippets/gen1-nextjs/next.config.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+   * `useIsPreviewing` uses a different react version while linking so here are resolving it
+   * this is not required when using the SDK directly
+   */
+  webpack: (config) => {
+    config.resolve.alias['react'] = path.resolve(
+      __dirname,
+      './node_modules/react'
+    );
+    return config;
+  },
```
Comment (sidmohanty11) [inline packages/sdks/snippets/gen1-nextjs/next.config.js:19]: had to add,
```
 "installConfig": {
    "hoistingLimits": "workspaces"
  },
```
and alias `react` package to the one present in the current `node_modules` as the `userIsPreviewing` hook was referencing a `useState` from a different version of React used by `@builder.io/react` causing issues

File: packages/sdks/snippets/gen1-remix/package.json [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    "dev": "remix vite:dev",
+    "lint": "eslint --ignore-path .gitignore --cache --cache-location ./node_modules/.cache/eslint .",
+    "start": "remix-serve ./build/server/index.js",
+    "typecheck": "tsc",
+    "test": "SERVER_NAME=gen1-remix yarn nx test:snippet @sdk/tests",
+    "serve": "remix-serve ./build/server/index.js"
+  },
+  "dependencies": {
+    "@builder.io/react": "workspace:*",
+    "@emotion/core": "10.0.17",
```
Comment (sidmohanty11) [inline packages/sdks/snippets/gen1-remix/package.json:17]: had to install `@emotion/core` as a dependency here as it was not able to correctly bundle the `@emotion/core` from `@builder.io/react` linked package [was trying to use the CJS build], but we would not need this when we are installing `@builder.io/react` as a package

Remix app deployed: https://github.com/sidmohanty11/builder-remix-gen1-example

File: packages/sdks/snippets/gen1-remix/package.json [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    "@remix-run/node": "^2.9.2",
+    "@remix-run/react": "^2.9.2",
+    "@remix-run/serve": "^2.9.2",
+    "isbot": "^4.1.0",
```
Comment (sidmohanty11) [inline packages/sdks/snippets/gen1-remix/package.json:21]: have used latest remix here (v2)

File: packages/sdks/snippets/gen1-remix/.eslintrc.cjs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    commonjs: true,
+    es6: true,
+  },
+  ignorePatterns: ['!**/.server', '!**/.client'],
+
+  // Base config
+  extends: ['eslint:recommended'],
+  rules: {
+    'import/no-unresolved': ['error', { ignore: ['@builder.io/react'] }],
+  },
```
Comment (sidmohanty11) [inline packages/sdks/snippets/gen1-remix/.eslintrc.cjs:28]: CI was not happy so added this, but locally couldn't see any errors

File: packages/sdks/snippets/gen1-remix/package.json (hunk 3 above)
Comment (samijaber) [reply packages/sdks/snippets/gen1-remix/package.json:17]: can you add this as a note in the README. it's an important gotcha for us maintaining it.

**Review by samijaber — APPROVED** [review]

File: packages/sdks/snippets/gen1-remix/package.json (hunk 3 above)
Comment (sidmohanty11) [reply packages/sdks/snippets/gen1-remix/package.json:17]: added

**Pull request merged**